src/data/value/mod.rs

```diff
@@ -75,26 +77,19 @@ impl PartialOrd<Value> for Value {
 
 impl Value {
     pub fn validate_type(&self, data_type: &DataType) -> Result<()> {
-        let valid = matches!(
-            (data_type, self),
-            (DataType::Boolean, Value::Bool(_))
-                | (DataType::Int, Value::I64(_))
-                | (DataType::Float, Value::F64(_))
-                | (DataType::Text, Value::Str(_))
-                | (DataType::Date, Value::Date(_))
-                | (DataType::Timestamp, Value::Timestamp(_))
-                | (DataType::Interval, Value::Interval(_))
-                | (DataType::UUID, Value::UUID(_))
-                | (DataType::Boolean, Value::Null)
-                | (DataType::Int, Value::Null)
-                | (DataType::Float, Value::Null)
-                | (DataType::Text, Value::Null)
-                | (DataType::Date, Value::Null)
-                | (DataType::Timestamp, Value::Null)
-                | (DataType::Time, Value::Null)
-                | (DataType::Interval, Value::Null)
-                | (DataType::UUID, Value::Null)
-        );
+        let valid = match self {
```

Comment:
Wow nice improvement!
Now we can catch the error when the new type is introduced.